src/components/cylc/tree/deltas.js

```diff
@@ -116,60 +82,96 @@ const UPDATED = {
   jobs: [createJobNode, 'updateJob']
 }
 
-/**
- * @typedef {Object} DeltasUpdated
- * @property {Array<Object>} familyProxies
- * @property {Array<Object>} taskProxies
- * @property {Array<Object>} jobs
- */
-
 /**
  * Deltas updated.
  *
  * @param updated {DeltasUpdated} updated
- * @param {CylcTree} tree
+ * @param {Workflow} workflow
+ * @param {Lookup} lookup
+ * @param {*} options
  */
-function applyDeltasUpdated (updated, tree) {
+function applyDeltasUpdated (updated, workflow, lookup, options) {
+  const result = {
+    errors: []
+  }
   Object.keys(UPDATED).forEach(updatedKey => {
     if (updated[updatedKey]) {
       updated[updatedKey].forEach(updatedData => {
         try {
-          const updateNodeFunction = UPDATED[updatedKey][0]
-          const treeFunction = UPDATED[updatedKey][1]
-          const node = updateNodeFunction(updatedData)
-          tree[treeFunction](node)
+          const existingData = lookup[updatedData.id]
+          if (!existingData) {
+            result.errors.push([
+              `Updated node [${updatedData.id}] not found in workflow lookup`,
+              updatedData,
+              workflow,
+              lookup
+            ])
+          } else {
+            const updateNodeFunction = UPDATED[updatedKey][0]
+            const treeFunction = UPDATED[updatedKey][1]
+            const node = updateNodeFunction(existingData)
+            CylcTree[treeFunction](node, workflow, options)
+          }
         } catch (error) {
-          // eslint-disable-next-line no-console
-          console.error('Error applying updated-delta, will continue processing the remaining data', error, updatedData)
-          store.dispatch('setAlert', new AlertModel('Error applying updated-delta, see browser console logs for more. Please reload your browser tab to retrieve the full flow state', null, 'error'))
+          result.errors.push([
+            'Error applying added-delta, see browser console logs for more. Please reload your browser tab to retrieve the full flow state',
+            error,
+            updatedData,
+            workflow,
+            lookup
+          ])
         }
       })
     }
   })
+  return result
 }
 
 /**
- * @typedef {Object} Deltas
- * @property {string} id
- * @property {boolean} shutdown
- * @property {?DeltasAdded} added
- * @property {?DeltasUpdated} updated
- * @property {?DeltasPruned} pruned
+ * Helper object used to iterate pruned deltas data.
  */
+const PRUNED = {
+  jobs: 'removeJob',
+  taskProxies: 'removeTaskProxy',
+  familyProxies: 'removeFamilyProxy'
+}
```

Comment:
I mentioned that to @dwsutherland yesterday on Element. But at the moment we are not handling when the workflow is removed in the tree. This code is being moved, but I agree we probably need to handle the removed. To be done in a follow-up, likely the #689 since I will be working on the deltas again.

Reply:
No probs, so long as it's on a ticket somewhere.